docs/azure/services/Azure_Functions.md

```diff
@@ -0,0 +1,109 @@
+
+The advice below does not consider any vulnerabilities that have been introduced into Azure Functions by user supplied code. Instead, it will focus on secure configuration practices when deploying Azure Functions. Any code provided to a function should be assessed before deployment.
+
+There are three Azure Function hosting plans, depending on which different security features are available:
+- Consumption Plan 
+- Premium Plan
+- Dedicated Plan
+- App Service Environment (ASE)
+- Kubernetes (Direct or Azure Arc)
+ASE, Kubernetes, and certain Dedicated Plan options allow further security controls to enforce isolation in the context of compute or networking configuration. This means that users will be able to run function apps in isolated, dedicated, and secure environments.
+
+## Assessment Notes
+### Networking Options
+The available network isolation options will depend on which of the three hosting options offered was used when creating a function. For example the Consumption, Premium and App Service plans run on a multi-tenant infrastructure. Consumption plan has the least number of options for network isolation, whilst App Service and Premium provide more robust controls to restrict network access. 
+
+Hosting a function in an ASE, deploys the function on a single-tenant infrastructure and offers full network isolation such as deploying it in a personal VNet. The network isolation options available for the above mentioned plans are summarised in a table [here] (https://learn.microsoft.com/en-gb/azure/azure-functions/functions-networking-options?tabs=azure-cli).
+
+#### Inbound Access Restrictions
+This option, which is available for all hosting plans, allows users to define a list of IP addresses which are allowed or denied access to a function. When no entries have been defined, an implicit “deny all” is applied.
+
+The following command will output a Function’s inbound access restrictions:
+```CLI
+az functionapp config access-restriction show -g ResourceGroup -n AppName
+```
+#### Outbound Traffic
+##### VNet Integration
+One of the ways to control outbound network access on a function is to set up a Virtual Network integration. This provides a function with the ability to communicate to resources within a dedicated VNet’s subnet. 
+
+This configuration can be found under the `Networking` blade or by running the following command:
+```CLI
+az functionapp vnet-integration list --name --resource-group
+```
+
+Once the VNet integration has been configured, the function will not by default have access to Azure resources. This would require a specific configuration following the change to ensure communication is possible with other internal resources.
```

Comment:
On another review the original terminology wasn't clear on whether function apps by default when integrated do not have inbound or do not have outbound connectivity to other Azure resources in the VNet. I would assume they do have outbound default connectivity from function -> resource but inbound from resource -> function is not allowed and would vary due to the regional or gateway-required integration. Worth clarifying that bit.